api/capi/include/nnstreamer.h

```diff
@@ -600,156 +606,131 @@ int ml_pipeline_valve_set_open (ml_pipeline_valve_h valve_handle, bool open);
- * @remarks If one of given property name does not exist, error code returns without fetching any properties.
  * @remarks The callers is responsible for freeing the allocated memory by calling g_free().
- * @param[in] elm_h The target element handle.
- * @param[in] first_property_name The name of first property to get.
- * @param[in] ... A varying number of name/return location pairs, finished by NULL.
+ * @param[in] elem_h The target element handle.
+ * @param[in] property_name The name of property to get.
+ * @param[out] value The value of specific property.
  * @return @c 0 on success. Otherwise a negative error value.
- * @retval #ML_ERROR_NONE Successful
+ * @retval #ML_ERROR_NONE Successful.
  * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
  * @retval #ML_ERROR_INVALID_PARAMETER Given parameter is invalid or given property name does not exist.
  *
- * Here is an example of the usage:
  * @code
  * ml_pipeline_h handle;
  * ml_pipeline_element_h elem_h;
  * gchar *pipeline;
  * gchar *ret_mode = NULL;
- * gboolean ret_silent;
  * int status;
  *
  * // Pipeline description
  * pipeline = g_strdup("videotestsrc ! video/x-raw,format=RGB,width=640,height=480 ! videorate max-rate=1 ! " \
- * "tensor_converter ! tensor_mux name=mux ! tensor_demux ! tensor_sink");
+ *  "tensor_converter ! tensor_mux name=mux ! tensor_demux ! tensor_sink");
  *
  * status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
  * if (status != ML_ERROR_NONE) {
- *   // handle error case
- *   goto error;
+ *  // handle error case
+ *  goto error;
  * }
  *
  * // Get the handle of target element
  * status = ml_pipeline_element_get_handle (handle, "mux", &elem_h);
  * if (status != ML_ERROR_NONE) {
- *   // handle error case
- *   goto error;
+ *  // handle error case
+ *  goto error;
  * }
  *
  * // Get single property of target element
- * status = ml_pipeline_element_get_property (elem_h, "sync_mode", &ret_mode, NULL);
+ * status = ml_pipeline_element_get_property (elem_h, "sync_mode", &ret_mode);
  * if (status != ML_ERROR_NONE) {
- *   // handle error case
- *   goto error;
+ *  // handle error case
+ *  goto error;
  * }
  * g_assert_true (g_strcmp0 ("slowest", ret_mode) == 0);
  * g_free (ret_mode);
  *
- * // Get multiple properties of target element
- * status = ml_pipeline_element_get_property (elem_h, "sync_mode", &ret_mode,
- *   "silent", &ret_silent, NULL);
- * if (status != ML_ERROR_NONE) {
- *   // handle error case
- *   goto error;
- * }
- * g_free (ret_mode);
- *
  * error:
- * ml_pipeline_element_release_handle (elem_h);
- * ml_pipeline_destroy (handle);
- * g_free(pipeline);
+ *  ml_pipeline_element_release_handle (elem_h);
+ *  ml_pipeline_destroy (handle);
+ *  g_free(pipeline);
  * @endcode
  */
-int ml_pipeline_element_get_property (ml_pipeline_element_h elm_h, const char *first_property_name, ...);
+int ml_pipeline_element_get_property (ml_pipeline_element_h elem_h, const char *property_name, ml_value_type value);
```

Comment:
Q: is it possible to set void * type ?
int ml_pipeline_element_get_property (ml_pipeline_element_h elem_h, const char *property_name, void *value);
int ml_pipeline_element_set_property (ml_pipeline_element_h elem_h, const char *property_name, const void *value);

Reply:
Yes, it's possible. In this case, developers should explicitly cast a primitive value as below.

```cpp
// API Header
int ml_pipeline_element_set_property (ml_pipeline_element_h elem_h, const char *property_name, const void *value);

// Developer's code
status = ml_pipeline_element_set_property(valve_h, "drop", (const void*)TRUE);
status = ml_pipeline_element_set_property(selector_h, "sync-mode", (const void*)1);
```

`g_object_set_data()` also has the same type-casting problem so it uses `GINT_TO_POINTER` macro as below.
```cpp
// Link: https://github.com/GNOME/glib/blob/mainline/gio/gio-tool-mount.c#L198
g_object_set_data (G_OBJECT (op), "state", GINT_TO_POINTER (MOUNT_OP_ABORTED));
```